### PR TITLE
fix: Store GEDCOM FAM.NCHI (number of children) on import/export

### LINKED ISSUE #490: GEDCOM import: NCHI tag (number of children) not stored
## Summary

The NCHI (Number of Children) GEDCOM tag is parsed but silently dropped. NCHI can differ from the actual CHIL count and represents the known number of children, including those not individually recorded.

### Location
`go-glx/gedcom_family.go`

_Migrated from todo.md — Import Gaps_

### PR DESCRIPTION
**Pull request by isaacschepp**
## What and why

The GEDCOM `FAM.NCHI` (number of children) tag was parsed during import but silently dropped — it fell through `convertFamily`'s `default` case, which only preserves underscore-prefixed extension tags. NCHI records the *known* number of children of a couple, which can legitimately differ from the count of individually-linked `CHIL` pointers (some children may not be recorded as separate persons).

This change stores NCHI as a new `number_of_children` relationship property (integer) on the marriage relationship and round-trips it back to `FAM.NCHI` on export.

Details:

- **Vocabulary**: registers `number_of_children` (`gedcom: "NCHI"`, `value_type: integer`) in the `relationship-properties` standard vocabulary. No JSON-schema change needed — the schema already accepts arbitrary property definitions and the `integer` value type.
- **Indices**: adds a `RelationshipProperties` map to both `GEDCOMIndex` (import) and `ExportIndex` (export), populated from the vocabulary's `gedcom` mappings exactly like the existing person/event/citation/... property indices.
- **Import** (`gedcom_family.go`): captures the NCHI subrecord in the existing two-pass loop and stores it on the relationship via `applyNumberOfChildren`. The payload is parsed to an `int` so it validates against the property's `integer` type. A non-numeric payload (e.g. `1 NCHI many`) is preserved verbatim with a warning rather than dropped. NCHI on a spouseless FAM (no relationship to attach to) warns instead of disappearing silently — matching the existing extension-tag handling in that branch.
- **Export** (`gedcom_export_family.go`): emits `1 NCHI <n>` from the property when the loaded vocabulary maps it to a GEDCOM tag.

## Related issues

Closes #490

## Review focus

- Storing the value as `int` (not the raw string) so it validates against `value_type: integer` — see `applyNumberOfChildren` and the new `TestImportFamilyNCHI_StoredOnRelationship` which asserts validation produces no `number_of_children` warning.
- The `RelationshipProperties` index additions mirror the existing property-index pattern on both import and export.
- INDI-level `NCHI` (a *person* attribute) is intentionally out of scope — it is unaffected and still handled by the individual converter's generic fallback.

## Testing

- `go test ./go-glx/` — all pass, including 5 new tests: import (stored as int + validates clean), non-numeric-preserved-with-warning, no-spouse-warns, export, and full import→export→re-import roundtrip.
- `go vet ./go-glx/`, `gofmt` clean.
- Repo `golangci-lint` (via lefthook pre-commit): **0 issues**.
- `node specification/validate-schemas.mjs` — all schemas valid.

## Breaking changes

None. Adds a new optional vocabulary property; existing archives are unaffected.

### CODE REVIEW

File: go-glx/gedcom_export_family.go [hunk 1]
```diff
@@ -365,6 +371,45 @@ func exportFamily(family *ExportFamily, expCtx *ExportContext) *GEDCOMRecord {
...
+// relationship's number_of_children property, when the property is set and the
+// loaded vocabulary maps it to a GEDCOM tag.
+func exportNumberOfChildren(rel *Relationship, expCtx *ExportContext) *GEDCOMRecord {
+	gedcomTag, ok := expCtx.ExportIndex.RelationshipProperties[RelationshipPropertyNumberOfChildren]
+	if !ok || gedcomTag == "" {
+		return nil
+	}
+
+	value := formatCountProperty(rel.Properties[RelationshipPropertyNumberOfChildren])
+	if value == "" {
```
[the remark this answers is not in the document]
Comment (isaacschepp) [reply go-glx/gedcom_export_family.go]: Addressed in cfdb79d.

Export is now vocabulary-driven and symmetric with import. `exportMappedRelationshipProperties` iterates the relationship's `Properties` and emits a GEDCOM subrecord for any key the loaded vocabulary maps to a tag (looked up via `ExportIndex.RelationshipProperties`), exactly mirroring how import resolves the property key via `GEDCOMIndex.RelationshipProperties` in `applyNumberOfChildren`. Both indices are built from the same `glx.RelationshipProperties` vocabulary (loaded once via `LoadStandardVocabulariesIntoGLX`), so a custom vocabulary that maps `NCHI` to a non-standard key now round-trips: import stores under that key and export reads it back via the same mapping.

Added a regression test (`TestExportFamily_NumberOfChildren_CustomVocabularyKey`) that maps `NCHI` to a custom `child_count` key and asserts export still emits `1 NCHI 4`. While here, an adversarial review pass also surfaced and fixed two related edge cases: `formatCountProperty` now renders `float64` losslessly (`3.5` → `"3.5"`) instead of silently truncating, and an empty `NCHI` payload now warns for parity with the non-numeric branch — with tests for `NCHI 0`, duplicate `NCHI`, empty `NCHI`, and a GEDCOM 7.0 round-trip.

**Comment by isaacschepp** [comment]
@copilot resolve the merge conflicts in this pull request

**Review by noahbjohnson — APPROVED** [review]
